app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/splash/SplashViewModel.kt

```diff
@@ -3,34 +3,74 @@ package com.kks.nimblesurveyjetpackcompose.viewmodel.splash
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.kks.nimblesurveyjetpackcompose.di.IoDispatcher
+import com.kks.nimblesurveyjetpackcompose.model.ErrorModel
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
 import com.kks.nimblesurveyjetpackcompose.repo.login.LoginRepo
+import com.kks.nimblesurveyjetpackcompose.util.extensions.mapError
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 var SPLASH_TIME = 2000L
 
-@Suppress("UnusedPrivateMember")
 @HiltViewModel
 class SplashViewModel @Inject constructor(
     private val loginRepo: LoginRepo,
     @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : ViewModel() {
-
+    private val _shouldShowLoading = MutableStateFlow(false)
     private val _shouldNavigateToLogin = MutableStateFlow(false)
+    private val _isLoginSuccess = MutableStateFlow(false)
+    private var _hasError = MutableStateFlow(ErrorModel())
+
+    val hasError: StateFlow<ErrorModel>
+        get() = _hasError.asStateFlow()
+
+    fun resetError() {
+        _hasError.value = ErrorModel()
+    }
 
     val shouldNavigateToLogin: StateFlow<Boolean>
         get() = _shouldNavigateToLogin.asStateFlow()
 
+    val shouldShowLoading: StateFlow<Boolean>
+        get() = _shouldShowLoading.asStateFlow()
+
+    val isLoginSuccess: StateFlow<Boolean>
+        get() = _isLoginSuccess.asStateFlow()
+
     fun startTimerToNavigateToLogin() {
         viewModelScope.launch(ioDispatcher) {
             delay(SPLASH_TIME)
             _shouldNavigateToLogin.value = true
         }
     }
+
+    fun login(email: String, password: String) {
+        viewModelScope.launch(ioDispatcher) {
+            loginRepo.loginWithEmailAndPassword(email = email, password = password).collect {
+                when (it) {
+                    is ResourceState.Loading -> {
+                        _shouldShowLoading.value = true
+                        resetError()
+                    }
+                    is ResourceState.Success -> {
+                        _shouldShowLoading.value = false
+                        _isLoginSuccess.value = true
+                        resetError()
+                    }
+                    else -> {
+                        _shouldShowLoading.value = false
+                        mapError(it, _hasError)
```

Comment:
Then update this accordingly 😄 
```suggestion
                        it.mapError()?.let { error -> _hasError.value = error }
```